larq_compute_engine/mlir/transforms/prepare_tf.cc

```diff
@@ -155,18 +164,42 @@ void PrepareLCE::runOnFunction() {
   // replaced with a single Conv op with dilation parameter.
   patterns.insert<ConvertTFDilatedConvOp<TF::Conv2DOp>>(ctx);
 
-  TFL::populateWithGenerated(ctx, patterns);
+  target_arm::populateWithGenerated(ctx, patterns);
+  applyPatternsAndFoldGreedily(func, patterns);
+}
+
+template <>
+void PrepareLCE<LCETarget::XCORE>::runOnFunction() {
+  OwningRewritePatternList patterns;
+  auto* ctx = &getContext();
+  auto func = getFunction();
+
+  // This pattern will try to identify and optimize for dilated convolution.
+  // e.g. Patterns like "SpaceToBatchND -> Conv2D -> BatchToSpaceND" will be
+  // replaced with a single Conv op with dilation parameter.
+  patterns.insert<ConvertTFDilatedConvOp<TF::Conv2DOp>>(ctx);
+
+  target_other::populateWithGenerated(ctx, patterns);
   applyPatternsAndFoldGreedily(func, patterns);
 }
 
 }  // namespace
 
 // Creates an instance of the TensorFlow dialect PrepareLCE pass.
-std::unique_ptr<OperationPass<FuncOp>> CreatePrepareLCEPass() {
-  return std::make_unique<PrepareLCE>();
+std::unique_ptr<OperationPass<FuncOp>> CreatePrepareLCEPass(
+    const LCETarget target) {
+  if (target == LCETarget::ARM) {
+    return std::make_unique<PrepareLCE<LCETarget::ARM>>();
+  } else {
+    return std::make_unique<PrepareLCE<LCETarget::XCORE>>();
+  }
 }
 
-static PassRegistration<PrepareLCE> pass("tfl-prepare-lce", "Inject LCE Ops");
+static PassRegistration<PrepareLCE<LCETarget::ARM>> pass_target_arm(
+    "tfl-prepare-lce-target-arm", "Inject LCE Ops, for ARM targets");
+
+static PassRegistration<PrepareLCE<LCETarget::XCORE>> pass_target_xcore(
+    "tfl-prepare-lce-target-xcore", "Inject LCE Ops, for XCORE targets");
 
```

Comment:
Same comment as above, why do we need multiple passes instead of options to a single pass